### ISSUE TITLE
Use before assignment not detected

### ISSUE THREAD

**Issue by Hibou57** [PyCQA/pylint#1807]
### Steps to reproduce

Check this with Pylint:

    def test(a):
        if a == 1:
            b = 2
        return 3 if b else 4

### Current behavior

Does not warn `b` is used without (very probably) being assigned (more below).

### Expected behavior

Opinions may vary, still here is mine: I know too many false alerts is at the detriment of alerts in the large. I know there use to be a Borland C++ compiler triggering annoying and noisy warnings when an assignment was conditional, which seems to be a case very close to this one. But there is only one condition, so clearly it can’t cover all cases. The probability `b` is not assigned, is high.

### pylint --version output

> pylint 1.7.4, 
> astroid 1.5.3
> Python 3.5.2 (default, Nov 23 2017, 16:37:01) 
> [GCC 5.4.0 20160609]

### Note

I’m not telling about it as a bug, rather a an hint.

**Comment by PCManticore** [comment]
This makes sense, but we could only detect it in a sane manner once we land control flow support in pylint.

**Comment by Hibou57** [comment]
@PCManticore , don’t worry too much, PyLint is already very clever even without it.

**Comment by jacobtylerwalls** [comment]
Duplicate of #1727 and #267.

**Comment by Pierre-Sassoulas** [comment]
Closing as a duplicate of #1727, it's old but the discussion happened elsewhere. (Thank you @jacobtylerwalls for helping us cleanup)
